thriftrw/spec/service.pyx

```diff
@@ -174,6 +175,39 @@ cdef class FunctionResultSpec(UnionTypeSpec):
             name, result_specs, allow_empty=(return_spec is None)
         )
 
+    cpdef object read_from(FunctionResultSpec self, ProtocolReader reader):
+        reader.read_struct_begin()
+
+        cdef dict kwargs = {}
+        cdef object val
+        cdef FieldSpec spec
+        cdef FieldHeader header
+
+        header = reader.read_field_begin()
+
+        # We use -1 to signify struct end due to cython constraints.
+        while header.type != -1:
+            spec = self._index.get((header.id, header.type), None)
+
+            if spec is None:
+                raise UnknownExceptionError(
```

Comment:
It's an unknown exception only if `header.id > 0`. If `header.id == 0`, a
value was returned for a `void` method and we can safely ignore it.